Contrib/AffiliationServer/revenue.fsx

```diff
@@ -0,0 +1,118 @@
+#r "nuget:fstoolkit.errorhandling"
+#r "nuget:nbitcoin"
+#r "../../WalletWasabi/bin/Debug/net7.0/WalletWasabi.dll"
+#r "System.Security.Cryptography.dll"
+
+open System.Security.Cryptography
+
+open System.Text
+open NBitcoin
+open NBitcoin.RPC
+open System
+open System.IO
+open Newtonsoft.Json
+open FsToolkit.ErrorHandling
+open WalletWasabi.Affiliation.Models
+open WalletWasabi.Affiliation.Serialization
+open WalletWasabi.Affiliation.Models.CoinJoinNotification
+
+module CommandLine =
+    let tryParseArgument (arg : string) =
+        match arg.Split([| '=' |]) with
+        | [| k; v |] -> Some (k, v)
+        | _ -> None
+
+    let getArg name =
+        Environment.GetCommandLineArgs()
+        |> Array.map tryParseArgument
+        |> Array.choose id
+        |> Array.tryFind (fun (argName, _) -> argName = name )
+        |> Option.map snd
+
+    let getArgOrDefault name defaultValue =
+        getArg name
+        |> Option.defaultValue defaultValue
+
+    let getArgOrFail name =
+        match getArg name with
+        | Some value -> value
+        | None -> failwith $"Mandatory argument {name} was not provided"
+
+module Signature =
+
+    let verifySignature (pubkey:byte[]) (message:byte[]) signature =
+        let ver = System.Security.Cryptography.ECDsa.Create()
+        let mutable bytesRead = 0
+        ver.ImportSubjectPublicKeyInfo(ReadOnlySpan(pubkey) , &bytesRead)
+        ver.VerifyData(message, signature, HashAlgorithmName.SHA256)
+
+    let verifyNotification affiliate pubkey (notification: CoinJoinNotificationRequest) =
+         let payload = Payload(Header.Create(affiliate), notification.Body)
+         verifySignature pubkey (payload.GetCanonicalSerialization()) notification.Signature
+
+open Signature
+open CommandLine
+
+let args = {|
+    rpcCredentials = RPCCredentialString.Parse (getArgOrFail "--connection")
+    network = Network.GetNetwork (getArgOrDefault "--network" "Main")
+    notificationsPath = getArgOrDefault "--path" "."
+    affiliate = getArgOrDefault "--affiliate" "trezor"
+    pubkey = getArgOrFail "--pubkey" |> Convert.FromHexString
+|}
+
+let rpc = RPCClient(args.rpcCredentials, args.network)
+let notificationFiles = Directory.EnumerateFiles (args.notificationsPath)
```

Comment:
This is a good point but I really don't know what extension it will be. In fact I still don't know if it will be a set of files or just one file with all the notifications. In any case I agree with the general comment but i don't know what is the best action. We could exclude `*.fsx` files...

Reply:
Ummm nop, i think your solution is better. I will do as you say.

Reply:
done.